templates/CMakeLists.txt.template

```diff
@@ -664,7 +665,7 @@
     "high performance general RPC framework"
     "<%text>${gRPC_CORE_VERSION}</%text>"
     "gpr openssl"
-    "-lgrpc -laddress_sorting -lcares -lz"
+    "-lgrpc -laddress_sorting -lupb -lcares -lz"
```

Comment:
I think this is wrong. upb is not a real dependency, it's not going to be statically linked into libgrpc (all the other libraries are explicit dependencies that can be installed separately, which is not the case of upb, as we know).

Reply:
I had the same thought before but I just tried to add upb because of address_sorting. Do you think address_sorting should be removed, too?

Reply:
Hmm it seems that this is necessary. Without this, cmake build fails. [log](https://source.cloud.google.com/results/invocations/492aef73-bc13-42e3-8346-b171899dc918/targets/github%2Fgrpc/tests)

```
g++ helloworld.pb.o helloworld.grpc.pb.o greeter_client.o -L/usr/local/lib `pkg-config --libs protobuf grpc++` -pthread -Wl,--no-as-needed -lgrpc++_reflection -Wl,--as-needed -ldl -o greeter_client
/usr/local/grpc/lib/libgrpc.a(backend_metric.cc.o): In function `grpc_core::ParseBackendMetricData(grpc_slice const&, grpc_core::Arena*)':
backend_metric.cc:(.text+0x19): undefined reference to `upb_alloc_global'
backend_metric.cc:(.text+0x25): undefined reference to `upb_arena_init'
...
```

Reply:
Fair enough.